actionpack/lib/action_controller/metal/request_forgery_protection.rb

```diff
@@ -32,29 +32,22 @@ class InvalidCrossOriginRequest < ActionControllerError #:nodoc:
   # response may be extracted. To prevent this, only XmlHttpRequest (known as XHR or
   # Ajax) requests are allowed to make requests for JavaScript responses.
   #
-  # It's important to remember that XML or JSON requests are also checked by default. If
-  # you're building an API or an SPA you could change forgery protection method in
-  # <tt>ApplicationController</tt> (by default: <tt>:exception</tt>):
+  # Subclasses of <tt>ActionController::Base</tt> are protected by default with the
+  # <tt>:exception</tt> strategy, which raises an
+  # <tt>ActionController::InvalidAuthenticityToken</tt> error on unverified requests.
+  #
+  # It's important to remember that XML or JSON requests are also checked by default.
+  # APIs may want to disable this behavior since they are typically designed to be
+  # state-less: that is, the request API client handles the session instead of Rails.
+  # One way to achieve this is to use the <tt>:null_session</tt> strategy instead,
+  # which allows unverified requests to be handled, but with an empty session:
   #
   #   class ApplicationController < ActionController::Base
-  #     protect_from_forgery unless: -> { request.format.json? }
+  #     protect_from_forgery with: :null_session
   #   end
   #
-  # It is generally safe to exclude XHR requests from CSRF protection
-  # (like the code snippet above does), because XHR requests can only be made from
-  # the same origin. Note however that any cross-origin third party domain
-  # allowed via {CORS}[https://en.wikipedia.org/wiki/Cross-origin_resource_sharing]
-  # will also be able to create XHR requests. Be sure to check your
-  # CORS configuration before disabling forgery protection for XHR.
-  #
-  # CSRF protection is turned on with the <tt>protect_from_forgery</tt> method.
```

Comment:
I was confused by this and then the documentation below that the default is `:null_session`(which it is on the method, but not really because we default to call it with `protect_from_forgery with: :exception` from initializers). 

Think we should bring that in line as well, and make `:exception` the default since it has always been default since 5.2 and before.

Reply:
Yeah, it's confusing that `:exception` is the default behaviour if you do nothing, but `:null_session` is the default behaviour if you call the method without no arguments.

Maybe we should change the latter to match the former, but it would need a deprecation cycle or a new framework default, which is beyond the scope of this pull request IMO. I'm happy to explore it as a followup.

Reply:
Oh not in this, in follow up 👍